File: app/src/main/java/com/emarsys/catseeker/network/NetworkClient.java
package com.emarsys.catseeker.network;

import android.util.Log;

import com.emarsys.catseeker.model.CatMessageModel;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class NetworkClient {

    // region Singleton
    private static final NetworkClient ourInstance = new NetworkClient();

    public static NetworkClient getInstance() {
        return ourInstance;
    }

    private NetworkClient() {
    }
// endregion

    // region Public methods
    public void postCatMessage(String message) {
        SelinaService selinaService = retrofit().create(SelinaService.class);
        Call<Void> call = selinaService.postCat(new CatMessageModel(message));
        call.enqueue(new Callback<Void>() {
            @Override
            public void onResponse(Call<Void> call, Response<Void> response) {
                Log.d("CAT", response.toString());
            }

            @Override
            public void onFailure(Call<Void> call, Throwable t) {
                Log.d("CAT", t.toString());
            }
        });
    }
// endregion

    // region Private methods
    private Retrofit retrofit() {
        return new Retrofit.Builder()
                .baseUrl("https://hooks.slack.com/")
                .addConverterFactory(GsonConverterFactory.create())
                .build();
    }
// endregion

}

File: app/src/main/kotlin/com/emarsys/catseeker/network/NetworkClient.kt
package com.emarsys.catseeker.network

import android.util.Log
import com.emarsys.catseeker.model.CatMessageModel
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object NetworkClient {

    val selinaService: SelinaService by lazy {
        Retrofit.Builder()
                .baseUrl("https://hooks.slack.com/")
                .addConverterFactory(GsonConverterFactory.create())
                .build()
                .create(SelinaService::class.java)
    }

    fun postCatMessage(message: String) {
        selinaService
                .postCat(CatMessageModel(message))
                .enqueue(object : Callback<Void> {
                    override fun onResponse(call: Call<Void>, response: Response<Void>) {
                        Log.d("CAT", "$response")
                    }

                    override fun onFailure(call: Call<Void>, t: Throwable) {
                        Log.d("CAT", "$t")
                    }
                })
    }
}